clustering/gke.py

```diff
@@ -0,0 +1,364 @@
+#!/usr/bin/python
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>
+
+DOCUMENTATION = '''
+---
+module: gke
+version_added: "2.0"
+short_description: Manage Google Container Engine (GKE) clusters.
+description:
+    - This module can manage Google Container Engine (GKE) clusters via the
+      GKE v1 API U(https://cloud.google.com/container-engine/reference/rest/)
+options:
+  name:
+    description:
+      - Name of the GKE cluster
+    required: true
+  description:
+    description:
+      - Description for the GKE cluster
+    required: false
+  node_count:
+    description:
+      - Description for the GKE cluster
+    required: false
+    default: 2
+  node_config_machine_type:
+    description:
+      - The nodeConfig machine type
+    required: false
+    default: "n1-standard-1"
+  node_config_disksize:
+    description:
+      - The nodeConfig disk size in GB
+    required: false
+    default: 100
+  node_config_scopes:
+    description:
+      - The nodeConfig list of scopes to set for each node
+    required: false
+    default: [
+        "https://www.googleapis.com/auth/compute",
+        "https://www.googleapis.com/auth/devstorage.read_only",
+        "https://www.googleapis.com/auth/logging.write",
+        "https://www.googleapis.com/auth/monitoring"
+        ]
+  node_config_metadata:
+    description:
+      - The nodeConfig metadata to add to each node's instance metadata
+    required: false
+  username:
+    description:
+      - The username for accessing the Kubernetes cluster
+    required: true
+    default: "admin"
+  password:
+    description:
+      - The password for accessing the Kubernetes cluster
+    required: true
+    default: "correct_horse_battery_staple"
```

Comment:
Thanks @roberthbailey - I'll set it to `required` but not provide a default. That will force users to set their own (and they can generate one randomly if desired).